File: java/lib/src/main/java/com/svix/EndpointListOptions.java
package com.svix;

public class EndpointListOptions extends ListOptions {
    @Override
    public EndpointListOptions iterator(final String iterator) {
        return (EndpointListOptions) super.iterator(iterator);
    }

    @Override
    public EndpointListOptions limit(final Integer limit) {
        return (EndpointListOptions) super.limit(limit);
    }
}

File: kotlin/lib/src/main/kotlin/EndpointListOptions.kt
package com.svix.kotlin

import com.svix.kotlin.models.Ordering

class EndpointListOptions : ListOptions() {
    var order: Ordering? = null

    fun order(order: Ordering) = apply { this.order = order }

    override fun iterator(iterator: String) = apply { super.iterator(iterator) }

    override fun limit(limit: Int) = apply { super.limit(limit) }
}
